cognite_toolkit/_cdf_tk/commands/auth.py

```diff
@@ -26,307 +30,364 @@
+                )
+                ToolGlobals.failed = True
+                return None
+            print("  [bold green]OK[/]")
+        except Exception:
             print(
-                "  [bold red]ERROR[/]: The service principal/application configured for this client does not have access to any projects."
+                "  [bold red]ERROR[/]: Not a valid authentication token. Check credentials (CDF_CLIENT_ID/CDF_CLIENT_SECRET or CDF_TOKEN)."
             )
             ToolGlobals.failed = True
             return None
-        projects = ""
-        projects = projects.join(f"  - {p.url_name}\n" for p in resp.projects)
-        print(projects[0:-1])
-    except Exception as e:
-        print(f"  [bold red]ERROR[/]: Failed to process project information from inspect()\n{e}")
-        ToolGlobals.failed = True
-        return None
-    print(f"[italic]Focusing on current project {auth_vars.project} only from here on.[/]")
-    print(
-        "Checking basic project and group manipulation access rights (projectsAcl: LIST, READ and groupsAcl: LIST, READ, CREATE, UPDATE, DELETE)..."
-    )
-    try:
-        ToolGlobals.verify_client(
-            capabilities={
-                "projectsAcl": ["LIST", "READ"],
-                "groupsAcl": ["LIST", "READ", "CREATE", "UPDATE", "DELETE"],
-            }
-        )
-        print("  [bold green]OK[/]")
-    except Exception:
+        try:
+            print("Checking projects that the service principal/application has access to...")
+            if len(resp.projects) == 0:
+                print(
+                    "  [bold red]ERROR[/]: The service principal/application configured for this client does not have access to any projects."
+                )
+                ToolGlobals.failed = True
+                return None
+            projects = ""
+            projects = projects.join(f"  - {p.url_name}\n" for p in resp.projects)
+            print(projects[0:-1])
+        except Exception as e:
+            print(f"  [bold red]ERROR[/]: Failed to process project information from inspect()\n{e}")
+            ToolGlobals.failed = True
+            return None
+        print(f"[italic]Focusing on current project {auth_vars.project} only from here on.[/]")
         print(
-            "  [bold yellow]WARNING[/]: The service principal/application configured for this client does not have the basic group write access rights."
+            "Checking basic project and group manipulation access rights (projectsAcl: LIST, READ and groupsAcl: LIST, READ, CREATE, UPDATE, DELETE)..."
         )
-        print("Checking basic group read access rights (projectsAcl: LIST, READ and groupsAcl: LIST, READ)...")
         try:
             ToolGlobals.verify_client(
                 capabilities={
                     "projectsAcl": ["LIST", "READ"],
-                    "groupsAcl": ["LIST", "READ"],
+                    "groupsAcl": ["LIST", "READ", "CREATE", "UPDATE", "DELETE"],
```

Comment:
Why did this become necessary now?

Reply:
You are right these are not necessary. My mistake.